Lib/email/feedparser.py

```diff
@@ -132,6 +353,75 @@ def __next__(self):
             raise StopIteration
         return line
 
+    def _get_dump(self, start_value:str|None = None):
+        _dump_destination = deque()
+        self._dump_destination = _dump_destination
+
+        if start_value:
+            _dump_destination.append(start_value)
+
+        # Flush our current _lines to _dump_destination
+        needs_more_data = False
+        for line in self:
+            if line is NeedMoreData:
+                needs_more_data = True
+                break
+            _dump_destination.append(line)
+
+        # Pull in more data, if we need more
+        if needs_more_data:
+            # Flush our partial, if we can
+            if self._partial and self._can_dump_partial(self._partial[0]):
+                _dump_destination.extend(self._partial)
```

Comment:
Should it be `self._partial` or `''.join(self._partial)` in this case? also, why are we only checking the first `partial[0]` content but still extend `_dump_destination` with the entire partial?

Reply:
Since we're only checking the partial, we only need to check the very first character of the partial (to see if it's a `-`), as well as the state of the parser. If it starts with a potential boundary (`-`), and we care about boundary matches, we assume we can't dump this at all, as we don't know the content of the rest of the line, and so we can't do the boundary matches. The rest of the line's content isn't looked at. It's worth noting that the partial will never contain a complete line here.

The extend() here just ensures we're definitely grabbing everything, since the partial could be many parts if it's a very long partial line.